qiskit_ibm_provider/ibm_backend.py

```diff
@@ -529,13 +534,28 @@ def _runtime_run(
     ) -> IBMCircuitJob:
         """Runs the runtime program and returns the corresponding job object"""
         hgp_name = self._instance or self.provider._get_hgp().name
+
+        session = self._session
+        if session:
+            if not session.active:
+                raise RuntimeError(f"The session {session.session_id} is closed.")
+            session_id = session.session_id
+            max_execution_time = session._max_time
+        else:
+            session_id = None
+            max_execution_time = None
+        start_session = session_id is None
```

Comment:
I think the logic might have to be altered a little because we don't want to have `start_session` as `True` if the user doesn't actually want to start a session.

right now something like this would still start a session:
```python
job = backend.run(circ)
```

Reply:
> 

You are right. I fixed that.